overlord/devicestate/remodel.go

```diff
@@ -238,6 +254,38 @@ func (rc *baseRemodelContext) SystemMode() string {
 	return rc.systemMode
 }
 
+func (rc *baseRemodelContext) setRecoverySystemLabel(label string) {
+	rc.recoverySystemLabel = label
+}
+
+func (rc *baseRemodelContext) postFinish() error {
+	if rc.model.Grade() == asserts.ModelGradeUnset {
+		// nothing special for non-UC20 systems
+		return nil
+	}
+	if rc.recoverySystemLabel == "" {
+		return fmt.Errorf("internal error: recovery system label is unset during remodel finish")
+	}
+	// for UC20 systems we need record the fact that a new model is used for
+	// booting and consider a new recovery system as as seeded
+	oldDeviceContext := rc.GroundContext()
+	newDeviceContext := &rc.groundDeviceContext
```

Comment:
this was never accessed like this so far. I wonder if for clarification we should add a comment where it is defined:

```
type baseRemodelContext struct {
        // groundDeviceContext will carry the new device model
	groundDeviceContext
```